server/src/main/java/org/elasticsearch/search/fetch/subphase/FetchSourceContext.java

```diff
@@ -84,12 +121,48 @@ public String[] excludes() {
         return this.excludes;
     }
 
+    public boolean includeVectors() {
+        return this.includeVectors;
+    }
+
     public boolean hasFilter() {
-        return this.includes.length > 0 || this.excludes.length > 0;
+        return this.includes.length > 0 || this.excludes.length > 0 || this.includeVectors == Boolean.FALSE;
     }
 
-    public SourceFilter filter() {
-        return new SourceFilter(includes, excludes);
+    public SourceFilter filter(MappingLookup mappingLookup) {
+        if (includeVectors()) {
+            return new SourceFilter(includes, excludes);
+        } else {
+            if (mappingLookup == null) {
+                throw new IllegalArgumentException("MappingLookup must not be null when filtering vectors");
+            }
+            String[] excludeFields = excludes();
+            String[] inferenceFields = mappingLookup.inferenceFields()
+                .keySet()
+                .stream()
+                .map(s -> s + ".inference.chunks.embedding")
```

Comment:
I think we should remove the chunked texts as well. Chunking is an implementation detail that I'd say goes in pair with the embeddings.

The "unchunked" text is stored in .text so we should be safe here.

Reply:
On an implementation note, you could use SemanticTextField.getInferenceFieldName(s) to get the field name to exclude.

Reply:
We want to keep the chunk text though, so what Samiul is doing here is correct. We just need to add a method to `InferenceFieldMapper` so this field name isn't hard-coded.

Reply:
> I think we should bring up this point with a larger audience to get a more informed decision on the default behavior.

Let's get feedback there - I heard the opposite as to completely hide the details. They can easily get the chunks using include_vectors.

Reply:
My two cents: I would only return the input text, and allow the chunks to be returned when people send in `include_vectors` explicitly.

Reply:
After investigating how to include vectors by default for reindex and update-by-query (see [this comment](https://github.com/elastic/elasticsearch/pull/112319#issuecomment-2346987612)), I changed my opinion here. We should only return the input text.

This enables us to more easily support reindexing with an inference model switch in a follow-up PR. We would just need to set `include_vectors` to null in the reindex request's underlying search request. The resulting filtered `semantic_text` output could be merged with any other `semantic_text` field mapping, regardless of the model used.